From: Enrico Tassi <gareuselesinge@debian.org>
Date: Mon, 26 Aug 2013 12:13:20 +0700
Subject: posix: port getcwd to Hurd.

* lposix.c (Pgetcwd): Use get_current_dir_name function on Hurd.

Signed-off-by: Gary V. Vaughan <gary@gnu.org>
---
 lposix.c | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/lposix.c b/lposix.c
index 5f3de33..469a094 100644
--- a/lposix.c
+++ b/lposix.c
@@ -673,6 +673,16 @@ Current working directory for this process.
 */
 static int Pgetcwd(lua_State *L)
 {
+#ifdef __GNU__
+	char *b = get_current_dir_name();
+	if (b != NULL) {
+		lua_pushstring(L, b);
+		return 1;
+	} else {
+		/* we return the same error as below */
+		return pusherror(L, ".");
+	}
+#else
 	long size = pathconf(".", _PC_PATH_MAX);
 	void *ud;
 	lua_Alloc lalloc = lua_getallocf(L, &ud);
@@ -686,6 +696,7 @@ static int Pgetcwd(lua_State *L)
 		lua_pushstring(L, b);
 	lalloc(ud, b, (size_t)size + 1, 0);
 	return (ret == NULL) ? pusherror(L, ".") : 1;
+#endif
 }
 
 /***
